WordPress/Classes/ViewRelated/Tools/SettingsTextViewController.m

```diff
@@ -126,16 +226,59 @@ - (UIView *)tableView:(UITableView *)tableView viewForFooterInSection:(NSInteger
 }
 
 
-#pragma mark - UITextFieldDelegate Methods
+#pragma mark - Helpers
 
-- (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string
+- (void)dismissViewController
 {
-    NSRange newLineRange = [string rangeOfCharacterFromSet:[NSCharacterSet newlineCharacterSet]];    
-    if (newLineRange.location != NSNotFound) {
+    if (self.isModal) {
+        [self dismissViewControllerAnimated:YES completion:nil];
+    } else {
         [self.navigationController popViewControllerAnimated:YES];
     }
+}
+
+- (void)notifyValueDidChangeIfNeeded
+{
+    if (self.onValueChanged == nil || [self.textField.text isEqualToString:self.text]) {
+        return;
+    }
+    
+    self.onValueChanged(self.textField.text);
+    self.onValueChanged = nil;
```

Comment:
`onValueChanged` should only be called *once*. This is meant to prevent double call, in the scenario in which the VC is in a NavigationController stack, and gets dismissed thru the keyboard.

For simplicity reasons, instead of adding a boolean to track if it was called form the keyboard event (OR `pop` gesture), we're just cleaning up the reference.

I'll add a comment to clarify. Thanks!